toolkits/global/packages/global-forms/README.md

```diff
@@ -13,3 +15,265 @@ The `global-forms` component currently uses the `DEFAULT` brand only.
+                    "id": "your-email",
+                    "name": "your-email"
+                }
+            ]
+        }
+    ]
+}
+```
+
+Where you do wish to include a legend, HTML is permissable, meaning you can include a heading to reinforce the form and page structure:
+
+```
+"fieldsets": {
+    [
+        {
+            "legend": "<h2>My level 2 legend</h2>",
+            "fields": [
+                ...
+            ]
+        }
+    ]
+}
+```
+
+(**Note:** it is valid and conforming to use heading elements inside `<legend>`s.)
+
+### Fields
+
+Fields take the `type` property to determine which of the `view/fields` they represent. This property along with `label`, `id`, and `name` should be mandatory in terms of your model.
+
+A wide range of standard input/field attributes are supported. So, for example, if you wanted to include a `readonly` attribute on your text input, you would just include a property of the same name on the data:
+
+```
+{
+    "type": "text",
+    "label": "Your email",
+    "id": "your-email",
+    "name": "your-email",
+    "readonly": true
+}
+```
+
+The `hint` property includes supplementary text under the main `<label>` text but _inside_ the `<label>`, meaning it is automatically available to screen reader software.
+
+```
+"hint": "This will be the email you signed up using"
+```
+
+The `optional` property affects the field’s label, appending _“(optional)”_ to the label text.
+
+```
+"optional": true
+```
+
+### Errors
+
+Each field can have an `error` property. The presence of the property indicates the field is in an error state and the property value (a string) defines the error message.
+
+```
+{
+    "type": "text",
+    "label": "Your name",
+    "id": "your-name",
+    "name": "your-name",
+    "error": "That is not a name, it’s a number"
+}
+```
+
+Errors can be summarized using a top level `errorSummary` property (adjacent to the `fieldset` property). Each error in the errors array must point to the `id` of the respective input and repeat its `error` message:
+
+```
+"errorSummary": {
+    "id": "summary",
+    "title": "Here’s what’s wrong",
+    "errors": [
+        {
+            "id": "your-name-error",
+            "error": "That is not a name, it’s a number"
+        },
+        {
+            "id": "checkbox-terms-error",
+            "error": "You must agree to the terms!"
+        }
+    ]
+}
+```
+
+### Multiple choices
+
+Select elements supply their choices via an `options` property, which must be an array. The `selected` property is a Boolean:
+
+```
+"options": [
+    {
+        "label": "Horse",
+        "value": "Horse"
+    },
+    {
+        "label": "Platypus",
+        "value": "Platypus",
+        "selected": true
+    }
+]
+```
+
+Radios provide _their_ choices via an `inputs` array:
+
+```
+{
+    "type": "radios",
+    "label": "Animal",
+    "inputs": [
+        {
+            "label": "Horse",
+            "value": "Horse",
+            "name": "animal"
```

Comment:
Would this make sense to hoist the name property from the inputs items to the field properties. Less repetitive and maybe less prone to mistakes?

Reply:
@morgaan I attempted that but handlebars wasn't having it for some reason. I might have another try.

Reply:
"handlebars – there's probably a helper you need to write for that"

Reply:
@Heydon Have you tried `../name`? So far I remember you can use this `..` operator to climb up the context.

Reply:
@morgaan It is a bit tricky with partials but I got it working with `{{> radio name=../name }}`. Updated the **README** to match. https://github.com/springernature/frontend-toolkits/pull/743/commits/4080e47d08be4220d95554454978c8dde53522ec